Automattic/jetpack
# Add scheduled updates sync option

Context: p1712832877052689-slack-C01A60HCGUA
See also: https://github.com/Automattic/jetpack/pull/36849

Now, a sync option is saved every time the cron is updated, or a log is added.

## Proposed changes:

* Save the scheduled updates in an option that is automatically synched so that can be retrieved on WPCOM
* Fix: `PUT /logs` endpoint now return 404 if it can't find the scheduled update

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Jetpack:
* On Jetpack Beta > WordPress.com Features install `add/scheduled-updates-sync-2`
* From Calypso create or update scheduled updates.
* In your site `_cli` run `wp option get jetpack_scheduled_plugins_update`. Ensure your scheduled update is there.

WPCOM
* Open `wpsh`
* `switch_to_blog( _BLOG_ID_ )`
* `print_r( get_option( 'jetpack_scheduled_plugins_update' ) );`
* Ensure the option is synched

WPCOM 2
* Make sure you are running the watcher
* `wp run-scheduled-update --require=bin/scheduled-updates/run-scheduled-updates.php --blog-id=__ID__`
* Make sure the CLI reports `Scheduled update found on sync option.`
* `wp run-scheduled-update --require=bin/scheduled-updates/run-scheduled-updates.php --blog-id=__ID__ --no-dry-run` to run it (now the script is on dry-run by default)
